app/assets/javascripts/initializers/initializeCommentDropdown.js

```diff
@@ -113,6 +107,13 @@ function initializeCommentDropdown() {
       return;
     }
 
+    // Android native apps have enhanced sharing capabilities for Articles
+    const articleShowMoreClicked = button.id === 'article-show-more-button';
+    if (articleShowMoreClicked && Runtime.isNativeAndroid('shareText')) {
```

Comment:
Does this work for web/PWA users like #1524?

Reply:
At the moment it doesn't as this is only available in the native app environment, thank you for the follow up this is a good catch. I think it can be tweaked so it supports the PWA environment as well. I'll try out a few ideas and push another commit soon 👍

Reply:
After trying a few ideas I made it work but there's a few edge cases where this doesn't play well for all platforms. 

I'll be tackling PWA support and other mobile platforms (iOS currently doesn't have this enhanced experience either) in a separate PR since this refactor looks good for now.

I'll report updates on this over in #9962